### DISCUSSION TITLE
Xcode Cloud can't build projects that uses Composable Architecture v1.4.0

### DISCUSSION THREAD

**Discussion by darrarski** [pointfreeco/swift-composable-architecture#2573] [Q&A] [answered]
### Description

After updating an Xcode project to use Composable Architecture v1.4.0, Xcode Cloud can no longer build it. CI fails with the following errors:

> Target 'CasePathsMacros' must be enabled before it can be used.

> Target 'ComposableArchitectureMacros' must be enabled before it can be used.

The project does not use any of the newly introduced macros. It builds fine with Xcode Cloud when using previous releases of Composable Architecture.

I'm aware that third-party macros should be enabled in Xcode before use. I'm unsure how to set it up to work when using Xcode Cloud.

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

Xcode Cloud can build a project that depends on Composable Architecture

### Actual behavior

Xcode Cloud fails to build a project that depends on Composable Architecture

### Steps to reproduce

1. Create a new Xcode project.
2. Add Composable Architecture dependency (make sure you use v1.4.0).
3. Setup Xcode Cloud.
4. Run Xcode Cloud build.

### The Composable Architecture version information

1.4.0

### Destination operating system

iOS 17, macOS 14

### Xcode version information

Xcode 15.1 beta 2 (15C5042i)

**Comment by stephencelis** [comment, marked as answer]
@darrarski This is a general new step required by macros. A quick web search led me to this page with a few ways of addressing the error:

https://forums.swift.org/t/is-there-a-way-to-programmatically-allow-trust-the-compiler-plugin-to-run-from-the-command-line/65690

Command line default you can set:

```
defaults write com.apple.dt.Xcode IDESkipMacroFingerprintValidation -bool
```

Argument you can pass to `xcodebuild`

```
-skipMacroValidation
```

Since this isn't a bug I'm going to convert it to a discussion.

**Reply by darrarski** [reply]
Thanks for the feedback! Adding the following line to `ci_scripts/ci_post_clone.sh` solves the problem:

```
defaults write com.apple.dt.Xcode IDESkipMacroFingerprintValidation -bool YES
```
